src/v3/src/components/Form/Layout.tsx

```diff
@@ -43,7 +45,8 @@ const Layout: FunctionComponent<LayoutProps> = ({ uischema }) => {
     >
       {
         elements.map((element, index) => {
-          const elementKey = `${element.type}_${index}`;
+          const stepNameKeyPart = stepName ? `_${stepName}_` : '';
+          const elementKey = `${element.type}${stepNameKeyPart}${index}`;
```

Comment:
1. Is the underscore delimiter just an arbitrary choice? AFAICT this is going to result in something like `VerticalLayout_identify-with-password_0`
2. Do we still need the index? Index shouldn't be used in the key prop if possible... eslint would complain if the key was constructed inside the JSX. I guess the problem is that it's possible/likely we'll have more than one of `VerticalLayout` or `HorizontalLayout`... we can address it separately.
3. nit:
```suggestion
          const elementKey = [element.type, stepName, index].join('_');
```

Reply:
1. yep is it, I know it doesnt matter in the grand scheme of things (as long as it is unique) but do we care if it is delimited or not?
2. We sort of still need the index because of components rendered that do not have a name but same type, for example a view that has multiple informational text elements will throw a console error of duplicate keys since the types and step name will be the same. I still think we need some sort of unique identifier that can be set on each element that can be used at this layout layer when constructing a key to be certain it is unique.
3. Thank you 🙏